src/aosm/azext_aosm/deploy/pre_deploy.py

```diff
@@ -0,0 +1,366 @@
+# --------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT
+# License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------
+"""Contains class for deploying resources required by NFDs/NSDs via the SDK."""
+
+from knack.log import get_logger
+
+from azure.core import exceptions as azure_exceptions
+from azure.cli.core.azclierror import AzCLIError
+from azure.mgmt.resource.resources.v2022_09_01.models import ResourceGroup
+
+from azext_aosm.util.management_clients import ApiClients
+from azext_aosm.vendored_sdks.models import (
+    ArtifactStore,
+    ArtifactStoreType,
+    NetworkFunctionDefinitionGroup,
+    NetworkServiceDesignGroup,
+    Publisher,
+)
+from azext_aosm._configuration import Configuration, VNFConfiguration
+from azext_aosm.util.constants import PROV_STATE_SUCCEEDED
+
+logger = get_logger(__name__)
+
+
+class PreDeployerViaSDK:
```

Comment:
SDK. Also is "PreDeployerVia<thing>" the best description of what this is doing? I feel like something like "PreDeploymentValidator" or something would be a more fitting name given what it does. Or "PreDeploymentSetup" or something to reflect that it may also create things.

Reply:
I'm going to leave this. I think I want the ViaSDK in there (although it's caused confusion this time), to differentiate the things we deploy with arm and the things we deploy with the sdk.

I like PreDeploymentSetup but think PreDeploymentSetupViaSDK is too long, so going to leave as it is. 

If people hate it we can always rename later. I've updated some docstrings to hopefully provide more clarity.